Follow XDG Base Directory Specification

Freso opened the pull request:
Place EnhancedDiscord files under $XDG_DATA_HOME as specified by the [XDG Base Directory Specification version 0.7](https://specifications.freedesktop.org/basedir-spec/basedir-spec-0.7.html).

This allows LinuxED to be run from anywhere while still having its data files located in a single location and also means for users that there’s one fewer location where data files will be scattered around their file system.

Fixes https://github.com/Cr3atable/LinuxED/issues/2

Freso committed:
Follow XDG Base Directory Specification

Place EnhancedDiscord files under $XDG_DATA_HOME as specified by the
XDG Base Directory Specification version 0.7[1].

This allows LinuxED to be run from anywhere while still having its data
files located in a single location and also means for users that there’s
one fewer location where data files will be scattered around their file
system.

Fixes https://github.com/Cr3atable/LinuxED/issues/2

[1] https://specifications.freedesktop.org/basedir-spec/basedir-spec-0.7.html

Cr3atable commented:
I'd prefer for this to be optional.
There's also a merge conflict with the latest commit. I will merge once these two issues are fixed.

twnlink committed:
Merge branch 'master' into follow-base-directory-standard

twnlink committed:
Change how the menu works

Previously the menu (and some important variables) were not changeable mid script run.
This commit fixes that by moving some important variables and other things into the main loop.
This also fixes the issue with installing ED to XDG.

Cr3atable commented:
This PR is now merged however I would like to comment on how the new update works.
Rather than defaulting to the XDG Base Directory, the installer now provides the option to use it upon starting as shown in the following screenshot:
![Example of option](https://i-am.a-virg.in/N76BWsGZ)

If you use this option the installer will then automatically use the XDG Base Directory.
If there is an EnhancedDiscord installation within the XDG Base Directory on start, the installer will default to the XDG Base directory.
This allows users to optionally pick what directory they want to install EnhancedDiscord in and makes sure not to be too much of a hassle if you choose anything besides the default.

Thank you for your pull request and issue.

Freso commented on LinuxED.py:
```diff
@@ -29,22 +29,19 @@
+            dirpath = os.environ['XDG_DATA_HOME']
+            currentdir = False
+    elif 'HOME' in os.environ:
+        if os.path.exists(os.path.join(os.path.join(os.environ['HOME'], '.local', 'share'), "EnhancedDiscord")):
```
`os.path.join(os.path.join(…), …)` is superfluous. Just use
```python
         if os.path.exists(os.path.join(os.environ['HOME'], '.local', 'share', "EnhancedDiscord")):
```

Freso commented:
Thanks. :) I don’t agree with not making the XDG standard approach the default, but I’m not gonna bother you more about that now that it is at least an option. ;)
